pkg/core/stateroot/prometheus.go

```diff
@@ -1,20 +1,30 @@
 package stateroot
 
-import "github.com/prometheus/client_golang/prometheus"
+import (
+	"github.com/nspcc-dev/neo-go/pkg/util"
+	"github.com/prometheus/client_golang/prometheus"
+)
 
-// stateHeight prometheus metric.
-var stateHeight = prometheus.NewGauge(
+// stateHeight (local and validated) prometheus metric.
+var stateHeight = prometheus.NewGaugeVec(
 	prometheus.GaugeOpts{
-		Help:      "Current verified state height",
+		Help:      "Current local or verified state height",
 		Name:      "current_state_height",
 		Namespace: "neogo",
 	},
+	[]string{"description", "value"},
 )
 
 func init() {
 	prometheus.MustRegister(stateHeight)
 }
 
-func updateStateHeightMetric(sHeight uint32) {
-	stateHeight.Set(float64(sHeight))
+func updateStateHeightMetric(sHeight uint32, sRoot util.Uint256, verified bool) {
+	var label string
+	if verified {
+		label = "Verified stateroot"
+	} else {
+		label = "Local stateroot"
+	}
+	stateHeight.WithLabelValues(label, sRoot.StringLE()).Set(float64(sHeight))
```

Comment:
Have you tested or tried to look in the grafana at a big height?
I am a bit concerned about the possible memory consumption https://github.com/prometheus/prometheus/issues/2227#issuecomment-263205865

Reply:
Memory management issue is a blocker, we've got 1.7 M blocks already and this number only grows. Anything that eats memory with every block is a big problem.